File: BMI_v2/app/src/main/java/com/example/bmi_v2/Recipes.java
package com.example.bmi_v2;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class Recipes extends AppCompatActivity {

    EditText bmi;
    TextView resulttext;
    String calculation;
    Button send_button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recipes);

        bmi = findViewById(R.id.bmi);
        resulttext = findViewById(R.id.recipe);
        send_button = findViewById(R.id.send_button);

        send_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                provideRecipes();
            }
        });


        //initialize and assign variable
        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);

        //set Home Selected
        bottomNavigationView.setSelectedItemId(R.id.recipes);

        //perform itemselectedlistener
        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                switch (menuItem.getItemId()){
                    case R.id.bmi:
                        startActivity(new Intent(getApplicationContext(), Bmi.class));
                        overridePendingTransition(0,0);
                        return true;
                    case R.id.home:
                        startActivity(new Intent(getApplicationContext(),MainActivity.class));
                        overridePendingTransition(0,0);
                        return true;
                    case R.id.bmr:
                        startActivity(new Intent(getApplicationContext(), Bmr.class));
                        overridePendingTransition(0,0);
                        return true;
                    case R.id.recipes:
                        return true;
                }
                return false;
            }
        });
    }
    public void provideRecipes() {
        String S1 = bmi.getText().toString();

        if(TextUtils.isEmpty(S1)) {
            bmi.setError("Please provide your int Bmi value");
            return;
        }

        float bmiValue = Float.parseFloat(S1);
        String proposedDish;
        if (bmiValue < 18.5){
            proposedDish = "Small Kebab";
        }
        else if (bmiValue >= 18.5 && bmiValue <= 24.9) {
            proposedDish = "Normal Kebab";
        }
        else if (bmiValue >=25 && bmiValue <=29.9){
            proposedDish = "Big Kebab";
        }
        else {
            proposedDish = "KingSize kebab, twice";
        }

        calculation = "Proposed dish: \n\n" + proposedDish + "\n";
        resulttext.setText(calculation);
    }
}

File: kotlin/BMI_v3/app/src/main/java/com/example/bmi_v3/Recipes.kt
package com.example.bmi_v3

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_recipes.*

class Recipes : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recipes)

        send_button.setOnClickListener{ provideRecipes() }
        //initialize and assign variable
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottom_navigation)
        //set Home Selected
        bottomNavigationView.selectedItemId = R.id.recipes
        //perform itemselectedlistener
        bottomNavigationView.setOnNavigationItemSelectedListener(BottomNavigationView.OnNavigationItemSelectedListener { menuItem ->
            when (menuItem.itemId) {
                R.id.bmi -> {
                    startActivity(Intent(applicationContext, Bmi::class.java))
                    overridePendingTransition(0, 0)
                    return@OnNavigationItemSelectedListener true
                }
                R.id.home -> {
                    startActivity(Intent(applicationContext, MainActivity::class.java))
                    overridePendingTransition(0, 0)
                    return@OnNavigationItemSelectedListener true
                }
                R.id.bmr -> {
                    startActivity(Intent(applicationContext, Bmr::class.java))
                    overridePendingTransition(0, 0)
                    return@OnNavigationItemSelectedListener true
                }
                R.id.recipes -> return@OnNavigationItemSelectedListener true
                R.id.quiz -> {
                    startActivity(Intent(applicationContext, Quiz::class.java))
                    overridePendingTransition(0, 0)
                    return@OnNavigationItemSelectedListener true
                }
            }
            false
        })
    }

    private fun provideRecipes() {
        val s1 = bmi.text.toString()
        val calculation: String

        if (TextUtils.isEmpty(s1)) {
            bmi.error = "Please provide your int Bmi value"
            return
        }
        val bmiValue = s1.toFloat()
        val proposedDish: String

        proposedDish = when (bmiValue.toInt()){
            in 0..19 -> "Small Kebab"
            in 19..25 -> "Normal Kebab"
            in 25..30 -> "Big Kebab"
            else -> "KingSize kebab, twice"
        }
        calculation = "Proposed dish: \n\n$proposedDish\n"
        recipe.text=calculation    }
}